packages/backend/src/Controllers/Election/claimElectionController.ts

```diff
@@ -0,0 +1,45 @@
+import ServiceLocator from '../../ServiceLocator';
+import Logger from '../../Services/Logging/Logger';
+import { permissions } from '@equal-vote/star-vote-shared/domain_model/permissions';
+import { expectPermission, hashString } from "../controllerUtils";
+import { BadRequest, Unauthorized } from "@curveball/http-errors";
+import { IElectionRequest } from "../../IRequest";
+import { Response, NextFunction } from 'express';
+
+var ElectionsModel = ServiceLocator.electionsDb();
+
+const className = "election.Controllers";
+
+const claimElection = async (req: IElectionRequest, res: Response, next: NextFunction) => {
+    Logger.info(req, `${className}.claimElection ${req.election.election_id}`);
+    // temp_id will be verified against the election owner id to grant the owner role (even if we're logged in)
+    expectPermission(req.user_auth.roles, permissions.canClaimElection)
+
+    // check for no-op
+    if(req.election.owner_id == req.user.sub){
+        res.json({ success: true })
+        return;
+    }
+
+    // must be logged in
+    if(req.user.typ != 'ID'){
+        throw new Unauthorized("User does not have permissions: must be logged in");
+    }
+
+    // verify claim key
+    if(hashString(req.body.claim_key) != req.election.claim_key_hash){
+        throw new Unauthorized("User does not have permissions: claim_key mismatch");
+    }
+
+    var failMsg = "Failed to update Election";
+    req.election.owner_id = req.user.sub;
+    const updatedElection = await ElectionsModel.updateElection(req.election, req, `Transferring Ownership`);
+    if (!updatedElection) {
```

Comment:
I don't think this case is actually possible since in the model we are using `.takeFirstOrThrow()` which will throw an exception if there are no results not return null. Also, an aside, I think we should be separating 4xx from 5xx better.  Bad request should fail input validation or could be thrown in some cases where there is no expected object, but for other failures between server and db we should be throwing 5xx.

Reply:
Ah, that makes sense.

Sounds like I can just remove this clause then. If updateElection returns a NoResultError and I don't catch it, then I believe it will translate to a 5xx for the user

Reply:
That seems right - it means we're assuming something went wrong on our side if insert fails which seems reasonable since we should have already validated request and thrown 4xx upstream of this if there was an issue.